### ISSUE TITLE
A simple suggestion for a cinematic, pre-view camera, early work . . .

### ISSUE THREAD

**Issue by charlesmlamb** [godotengine/godot-proposals#3350]
### Describe the project you are working on

A barbarian game, 2.5D . . .

### Describe the problem or limitation you are having in your project

Well, I need to have some cinematics, been working on a reliable, cool preview environment, early test . .

![Skærmbillede (2284)](https://user-images.githubusercontent.com/83387915/134765903-5b94eaca-0eb2-4ac0-9003-f742bf3319c1.png)

And . .

![Skærmbillede (2288)](https://user-images.githubusercontent.com/83387915/134765909-403afebc-52e7-4d69-a457-8ff816c460fb.png)

To me, making cinematics is a work, that needs great respect, for the viewer, or public . . . So, I've been trying to make a respectable, also modest preview environment, similar to pro video - editing, also camera - crews . . . One might make a few variations, maybe have four crosses, in the middle, anyway I want to have a pro - vibe environment, to sort-of work in a comfortable, reliable environment . . .

Made the above, it's an idea for a new node, CinematicCamera3D, don't know what to call it . . . :OO . .

It'd just need to be two camera, where one is bit further back, and the second one doesn't well render the crosses, so one can preview it, sort-of 1-to-1, and also have some help, to set up the cinematics, might be about time Godot, given the amazing Godot 4 work, to get people a reliable, respectable camera - cinematic preview, in the cases, where it makes sense to add stuff, like cut - scenes, it's not always wise, to make drama, sort-of for pure sake, of it, my opinion . . .

Anyway, it gives Godot a bit more pro - vibe, sort-of a bit help, in the UX, if one wants to make cinematics, to keep it 'cool' or, not sure, respectable . . . Suggestion is, make a camera, that has something like that, so perhaps people that work at pro - level, high - quality work, will find engine a nicer place, to work . . . Not sure this is the best, tried the best, I could . . . 

Since, I'm not sure if this is well best, maybe there could be small discussion, could this be added and, if so, can it be made respectable or, clean . . <3 Thx . . .

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Like above, sort-of, add a camera, that gives a pro - editing or, cinematic recording ' experience ', without compromising <3 . . .

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Like above, I'm gonna use it, have some important stuff, need to make, or so . . That's how I work, don't think it can be made much better, it's something, to debate, or so . . . .

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Well, one can set it up manually, I think it'd make sense, as an improved camera, for respectable directors, or modest content creators . . .

### Is there a reason why this should be core and not an add-on in the asset library?

Well, thinking Godot 4 is getting so good, one can make almost cinematics, atm . . . So, why not make the best, most reliable version or, edition of a maybe cinematic camera, maybe someone has some ideas, etc . . . . <3

ps. Sorry about noob resolution, of images, gonna make it shine, as much as possible, that will come later . . .

**Comment by Calinou** [comment]
Have you tried the **Cinematic Preview** feature in the 3D viewport's **Perspective** menu?

**Comment by charlesmlamb** [comment]
I heard it was for sort-of seeing different cameras, it doesn't do this . . .

Well, imo. making drama, or ' fiction ' isn't for beginners, so the stuff I made isn't ' clean ', or sober enough, maybe someone could make this, so it fits Godot, in a modest, respectable way, I can make it, right now I'm just hoping for a discussion, how this could be done, in a respectable or, perhaps ' Godot ' way . . .

One could add more crosses, maybe five in middle, four on each side, help send a message, drama or, well 'bull - shit' isn't for beginners, create a supporting environment, however right now it's for discussion, if it is even possible . . . . <3

**Comment by charlesmlamb** [comment]
I think, one could make a 3 crosses on each side, four crosses in middle, and it'd work almost perfect, for Godot . . . Having a slider to set between lowest 3, 4, and 5 crosses, might make it respectable, imo. lowest is 3 crosses on side, 4 in the middle . . .

Mostly sure, need someone to look at it, and IF to add something like, HOW, with respect . . <3

**Comment by SIsilicon** [comment]
Could you summarize that? I've failed to understand what you're saying.

**Comment by charlesmlamb** [comment]
Most pro - recording crews have something like the above . . . A sort of preview, where there are reference points, for top-of-line movie production . . . Giving what movies are, atm., not sure I fully support that, if one wanted something close to that, it tends to look, like that . . .

![2021-09-25 1650](https://user-images.githubusercontent.com/83387915/134775646-0b55ca18-b766-4bbc-8417-e510e474d851.jpg)

I've been trying to make one, that fits my projects, there are others, used in high - end documentary, journalist work, some movies . . . These are overly 'commercial', but are examples, of how pro - recording stuff looks . . . The discussion is, could this be made in Godot, in a modest, competent way . . .

![2021-09-25 1651](https://user-images.githubusercontent.com/83387915/134775722-9c46df2b-3a76-472f-a970-97535730970d.jpg)

The examples are the best I have been able, to find . . . Atm, Godot can import animations, also PBR models, the engine can make cinematics, I've made this for myself, to feel 100 % comfortable, it's something that could be added, however was hoping for feedback, it's a suggestion, for allowing the making, of high - quality cinematics, in Godot . . . 

Another example, again this is for well commercial media, not really a fan, tbh . . . :)) . . 

![2021-09-25 1656](https://user-images.githubusercontent.com/83387915/134775861-e48c7f70-a4aa-42ca-a9fb-c0783556388b.png)

Anyway, I need to make some high - quality cinematics, so just gonna use what I have, might make it bit better . . It helps the person making the cinematic, better visualizes points, and zones, needed for ' optimizing ' the cinematics, not a goal in itself, I need for my own project, something quite amazing, also relevant, etc . . . 

It might be possible to make a ' Godot ' version, I think the suggestion, is fairly close . . . .

**Comment by Zireael07** [comment]
Those pictures really clear up what you meant by 'crosses on the side'. Yes, that could be useful indeed.

**Comment by charlesmlamb** [comment]
The debate I'm hoping for, is IF this can even be added, HOW, is there a good way, if not, it is prob. not possible . . . .

Other examples, it helps guide the 'cinematic' and, make sure actors, objects are in correct spaces, don't leave the cinematic, etc . . There's a reason these are in all top - level recording gear, it helps the artist, documentarist, journalist, or, so on . . . .

![2021-09-25 1701](https://user-images.githubusercontent.com/83387915/134776048-1b71677d-076b-4fd2-bbcf-0c4f341e545b.jpg)

One could also use circles, many cameras use them for focus purposes, but for top - level cameras, say for a movie, they have some reference grids, to help the director, so on, to make most of shot, or whatever . . . I'm NOT saying add some commercial trash, question is, can one make something, that would fit, with Godot, to help people needing a cinematic, or perhaps even many, for some important project, atm. I need that, it's just standard, at high level, in almost any recording area, documentaries, journalism, almost anything well done, one can find on tv, well, sort - of pro - stuff, for high - level productions . . Knowing, the media, it's not something necessarily to strive for, anyway this is my suggestion, for a ' Godot ' version, but, please debate, I'm not the only person here, and feedback only makes stuff better, thanks . . . .

![Skærmbillede (2289)](https://user-images.githubusercontent.com/83387915/134776172-ad48a391-b3e2-4a18-a710-2d3189836a83.png)

Also, it's prototype, made it in MS Paint :), gonna make it better, just feel comfortable, working this way, that's all . . .

**Comment by charlesmlamb** [comment]
Tbh, don't know why they use that, for my projects I needed the best, of the best, and that's it . . . Sort-of, from when cinema was classy, or respectable . . . Imo, when it all peaked, the balance between maybe innocence and, presentation, was the finest, maybe like Jurassic Park 1, or before that . . Old-school, respectable recording - gear, sometimes they use the stuff, to zoom, sort-of auto - focus, but that was not why they used this, I think it was to help make sure, the recording was almost pristine, or ' cool ', nice . . .

I'm gonna use it, my own stuff, bec. it makes sense in my projects, if I had to add this, I'd do it like this, to go full on, 4 crosses on side, 4 crosses in middle, or 5 . . That's when, you know your stuff, or well prob. can't look at that many crosses lol :DD . . . .

That's why, I THINK this might fit Godot, to send message, cinematics, drama, etc . . Is fine, but there's also some respect, or reflection involved, bef. you just ' make ' stuff, for no reason . . . Thx . . !

**Comment by charlesmlamb** [comment]
Sort-of, don't know if you remember T-Rex in Jurassic Park 1, it was sort-of obvious, it was put there, not too real, they didn't fool people, also they used puppets, from stuff, etc . . .

![2021-09-25 1742](https://user-images.githubusercontent.com/83387915/134777337-14e9007f-31fb-407d-b5c4-c10a483628ef.jpg)

They never ' conned ' people, you always knew, it's just a movie, or ' composite ' stuff, pictures . . . That's when I felt it was most honest, and recording gear from that time, looked like that, when the balance between fun and, 'manipulation' didn't get crossed, sort-of . . .

So, made it for myself, might be the ' best ' way to add this to Godot, esp. the 4 version, that's will prob. be ready, to make AA - AAA cinematics, in the right hands, or so . . . .

**Comment by charlesmlamb** [comment]
Ok, made a quick test, think I know why it's a good idea, for making stunning, if needed, cut-scenes, cinematics, etc . . .

3 minute video, with explaination . . .

>> https://www.youtube.com/watch?v=cHNqfKCc090

One sort of can make a sequence of still photos, that are appealing, or harmonious, like this . . . .

<img width="960" alt="Skærmbillede (2331)" src="https://user-images.githubusercontent.com/83387915/135708581-cfed11a2-f8b5-4fc5-83b5-f5f12f3ec640.png">

First key-frame, for the cinematic, image is balanced, harmonious, clean . . .

<img width="960" alt="Skærmbillede (2332)" src="https://user-images.githubusercontent.com/83387915/135708589-42d18fc8-2bb1-4dff-8878-44672f563272.png">

2nd shot, make sure it's a nice, clean image, etc . . . .

<img width="960" alt="Skærmbillede (2333)" src="https://user-images.githubusercontent.com/83387915/135708599-ea753645-0443-476b-a165-5859f8cd9691.png">

3rd image, again I make sure, it's a really nice image, captures the frame, in a nice way . . . THEORETICALLY, this would make an animation, of camera, key-frame based sort-of make an almost ' cinematic ', or pro - photo ' feeling ', also the center cross, makes sure stuff is always in center, of the screen, something the normal camera preview doesn't show . . . It's cheap, requires a single texture, I use two cameras, see video . . .

That's my suggestion, for improving the cinematics QoL, of Godot, one works in some single photos, then make camera interpolate between them, to get top - grade images, might be same, for the animation bezier curve, for the camera, esp. nice if one could work with focus, zoom, and other things, that's how I get a pro - cinematic software, for basically 0 dollars <3

I just make sure, that it's a sequence of images, that for each one, is a photographics master-piece, this should lead to some nice cut - scenes, not sure . . . . <3 <3

![2021-10-02 1005](https://user-images.githubusercontent.com/83387915/135708750-d5390ea8-b18a-4cdc-8dce-a130cadcc286.jpg)

![2021-10-02 1006](https://user-images.githubusercontent.com/83387915/135708754-973a6f0e-f57c-4c35-b71c-5e9666211d1a.jpg)

Super - cheap, right now Godot has not even, a center cross-point, or focus point, one is sort of left guessing, where the camera is focused, it's also difficult, bec. one has to set near-far DoF, or blur range, in environment settings, one MIGHT make an almost pro - camera, to help people make better cinematics, by adding that, to a special camera, so one animates one node, to get an ' optimal ' shot, it'd bas. make Godot 90 % pro, or close, for making cinematics, etc . . . Also, 0 dollars <3

**Comment by charlesmlamb** [comment]
It sort-of takes all the guesswork out, of making amazing, next - level cut-scenes, cinematics, just make sure that each keyframe, is a world - class photo, booom, best looking movie, cinematic, ever . . Thanks . . . .

**Comment by charlesmlamb** [comment]
Well, here's the point, I THINK all movie, documentary makers, sort-of think like this, one wants to draw the viewer to one part, of the screen, and have all elements ' set ' up, so that happens, to sort-of ' get ' an audience's attention, like this . . . .

<img width="640" alt="2021-10-02 1027" src="https://user-images.githubusercontent.com/83387915/135709296-7692f90c-9169-41ac-a1ff-97521e26af9f.png">

Or . . . .

<img width="640" alt="2021-10-02 1028" src="https://user-images.githubusercontent.com/83387915/135709300-d5ace8f5-e9d4-4d60-89f5-cb80f2d83a2b.png">

And, having a ' grid ', over a screen, helps make sure the cinematic is optimized, to get the viewer's attention, think that's why they use it, for top - of - line cinematic stuff, thanks <3 . . !
